src/util/ConfigManager/ConfigManager.h

```diff
@@ -287,5 +353,78 @@ class ConfigManager {
+
+  /*
+  @brief The implementation of `addValidator` and `addOptionValidator`.
+
+  @param addValidatorFunctionName The name of the function, that is calling this
+  implementation. So that people can easier identify them.
+  @param translationFunction Transforms the given `ConfigOptionProxy` in the
+  value form for the validator function.
+  @param validatorFunction Should return true, if the transformed
+  `ConfigOptionProxy` represent valid values. For example: There configuration
+  options were all set at run time, or contain numbers bigger than 10.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param parameterProxy Proxies for the configuration options, who will be
+  passed to the validator function as function arguments, after being
+  transformed. Will keep the same order.
+  */
+  template <typename TranslationFunction, typename ValidatorFunction,
+            isInstantiation<ConstConfigOptionProxy>... ValidatorParameter>
+  requires(std::invocable<TranslationFunction, const ValidatorParameter> &&
+           ...) &&
+          Validator<ValidatorFunction,
+                    std::invoke_result_t<TranslationFunction,
+                                         ValidatorParameter>...> &&
+          (sizeof...(ValidatorParameter) > 0)
+  void addValidatorImpl(std::string_view addValidatorFunctionName,
+                        TranslationFunction translationFunction,
+                        ValidatorFunction validatorFunction,
+                        std::string_view errorMessage,
+                        const ValidatorParameter... parameterProxy) {
+    // Check, if we contain all the configuration options, that were given us.
+    auto checkIfContainOption = [this, &addValidatorFunctionName]<typename T>(
+                                    ConstConfigOptionProxy<T> opt) {
+      if (!containsOption(opt.getConfigOption())) {
+        throw std::runtime_error(absl::StrCat(
+            "Error while adding validator with ", addValidatorFunctionName,
+            ": The given configuration "
+            "option '",
+            opt.getConfigOption().getIdentifier(),
+            "' is not contained in the configuration manager, or its sub "
+            "managers."));
+      }
+    };
+    (checkIfContainOption(parameterProxy), ...);
+
+    /*
+    Add a function wrapper to our list of validators, that calls the
+    `validatorFunction` with the config options and throws an exception, if the
+    `validatorFunction` returns false.
+    */
+    validators_.emplace_back([translationFunction, validatorFunction,
+                              errorMessage = std::string(errorMessage),
+                              parameterProxy...]() {
+      if (!std::invoke(validatorFunction,
+                       std::invoke(translationFunction, parameterProxy)...)) {
+        throw std::runtime_error(errorMessage);
```

Comment:
The error message should automatically include the name of the verified config option for easier usability.

Reply:
Done, see [fd603f8](https://github.com/ad-freiburg/qlever/pull/1058/commits/fd603f8f1975549cdf628d3f3216b330766fc2dc).